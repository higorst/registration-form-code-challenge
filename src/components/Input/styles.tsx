import styled from 'styled-components/native'
import { Dimensions, TextInputProps } from 'react-native'
import { Colors } from '../../styles/Colors'

export const InputText = styled.TextInput<TextInputProps>`
    /* component */
    padding: 10px;
    margin: 5px;
    margin-bottom: 10px;
    width: ${`${Dimensions.get('window').width * 0.9}px`};

    align-self: center;
    align-content: center;

    border-radius: 10px; 
    background-color: ${Colors.white};

    /* text */
    font-size: 16px;
`