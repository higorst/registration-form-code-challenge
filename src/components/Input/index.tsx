import React from 'react'
import { Constants } from '../../constants';
import { Colors } from '../../styles/Colors';
import { InputText } from './styles'

interface PropsInput {
    placeholder: string

    value: string
    KeyboardType?: string
    onChangeText?: any
    sending?: boolean
    onlyNumber?: boolean
}
function Input(props: PropsInput) {

    return (
        <InputText 
            onChangeText={(string: string) => props.onChangeText(string)}
            placeholderTextColor={Colors.gray}
            placeholder={props.placeholder}
            value={props.value}
            editable={!props.sending}
            autoCapitalize="words"
            clearButtonMode="always"
            keyboardType={props.onlyNumber ? 'number-pad' : 'default'}
            autoFocus
        />
    )
}

export default Input