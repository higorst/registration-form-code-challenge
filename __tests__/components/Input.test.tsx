import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../src/components/Input/index'

jest.useFakeTimers()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper')

describe('<Input />', () => {
    it('renders Input Component correctly', () => {
        const tree = renderer.create(
                <Input
                    value='Snapshot'
                    placeholder='Snapshot'
                    onChangeText={() => {}}
                    sending={false}
                />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});