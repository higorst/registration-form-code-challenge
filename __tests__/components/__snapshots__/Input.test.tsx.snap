// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders Input Component correctly 1`] = `
<TextInput
  allowFontScaling={true}
  autoCapitalize="words"
  clearButtonMode="always"
  editable={true}
  keyboardType="default"
  onChangeText={[Function]}
  placeholder="Snapshot"
  placeholderTextColor="#d0cfd0"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignSelf": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 10,
        "fontSize": 16,
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": 675,
      },
    ]
  }
  underlineColorAndroid="transparent"
  value="Snapshot"
/>
`;
